src/libponyrt/lang/socket.c

```diff
@@ -90,7 +90,7 @@ typedef struct
   struct sockaddr_storage addr;
 } ipaddress_t;
 
-static socklen_t address_length(ipaddress_t* ipaddr)
+PONY_API socklen_t address_length(ipaddress_t* ipaddr)
```

Comment:
If we want to expose this for FFI use in the standard library, it needs to get either a `pony_` or `ponyint_` prefix to its name, for cleanliness of the function namespace. `pony_` means it is a public API meant for use by third party code, whereas `ponyint_` means it is internal. Unless there is a good reason to make it public, we should prefer to keep it internal because public functions require us to treat any changes to them as breaking changes in Pony.

So, this function should end up named `ponyint_address_length`, or something else with that prefix.